src/NzbDrone.Core/RootFolders/RootFolderService.cs

```diff
@@ -129,6 +130,30 @@ public void Remove(int id)
             _rootFolderRepository.Delete(id);
         }
 
+        private String getTheTvIdFromNfo(DirectoryInfo di)
+        {
+            String returnValue = null;
+            var fileNfoName = Path.Combine(di.FullName, "tvshow.nfo");
+            if (_diskProvider.FileExists(fileNfoName))
+            {
+                try
+                {
+                    XmlDocument doc = new XmlDocument();
+                    doc.Load(File.Open(fileNfoName, FileMode.Open));
+                    XmlNode node = doc.DocumentElement.SelectSingleNode("/tvshow/id");
+                    if (node != null)
+                    {
+                        returnValue = node.InnerText;
```

Comment:
Why not return directly from here? This should also parse it to an integer to ensure it's valid.